src/app/content/components/Page/contentLinkHandler.ts

```diff
@@ -32,20 +32,31 @@ export type ContentLinkProp =
 
 export const reduceReferences = (document: Document, {references, currentPath}: ContentLinkProp) => {
   for (const reference of references) {
-    const path = content.getUrl(reference.params);
-    const search = content.getSearch && content.getSearch(reference.params);
-    const query = search ? `?${search}` : '';
-    const a = assertNotNull(
-      document.querySelector(`[href^='${reference.match}']`),
-      'references are created from hrefs');
-    const href = assertNotNull(a.getAttribute('href'), 'it was found by href value')
-      .replace(reference.match, toRelativeUrl(currentPath, path) + query);
-    a.setAttribute('href', href);
+    // references may contain PageReferenceMapError only if UNLIMITED_CONTENT is set to true
+    if (isPageReferenceMapError(reference)) {
```

Comment:
this function would be less intimidating if you split the two sections out into separate little helpers